docs/build/guides/basics/automate-reset-data.mdx

```diff
@@ -0,0 +1,460 @@
+
+You can check current reset dates [here](../../../learn/fundamentals/networks.mdx#testnet-and-futurenet-data-reset).
+
+## Why resets are important
+
+1. **Clean Slate:** Regular resets ensure that both Testnet and Futurenet provide a clean environment for testing. This helps in avoiding complications arising from old data or configurations.
+2. **Performance:** Over time, test environments can accumulate a lot of data, which can slow down performance. Resets help in maintaining optimal performance.
+3. **Protocol Updates:** Introducing new features or protocol changes often requires a reset to ensure compatibility and stability.
+4. **Development Cycles:** Aligning with development cycles allows developers to plan their testing phases and ensures they have a reliable environment for their work.
+
+## Data automation on Testnet and Futurenet
+
+Automating blockchain state on Stellar's Testnet and Futurenet can streamline development workflows, ensuring that you can consistently test and validate your applications in these environments.
+
+### Code walkthrough
+
+### Prerequisites:
+
+- [Node.js](https://nodejs.org/en) and `npm` installed.
+- Stellar SDK for [JavaScript](https://www.npmjs.com/package/@stellar/stellar-sdk) and `fs` installed
+
+### Code
+
+```javascript
+import * as StellarSdk from "@stellar/stellar-sdk";
+import fs from "fs";
+
+// This Functions Submits a transactions and then polls for its status until a timeout is reached.
+async function yeetTx(feedback, server) {
+  try {
+    if (feedback.status !== "PENDING") {
+      throw feedback;
+    }
+
+    let status;
+    let attempts = 0;
+    while (attempts++ < 20) {
+      // Increased max attempts
+      const tmpStatus = await server.getTransaction(feedback.hash);
+      switch (tmpStatus.status) {
+        case "FAILED":
+          throw tmpStatus;
+        case "NOT_FOUND":
+          await sleep(1000); // Increased sleep time
+          continue;
+        case "SUCCESS":
+          status = tmpStatus;
+          break;
+      }
+      if (status) break;
+    }
+
+    if (attempts >= 20 || !status) {
+      // Updated max attempts check
+      throw new Error(
+        `Failed to find transaction ${feedback.hash} after ${attempts} attempts.`,
+      );
+    }
+
+    return status;
+    // console.log("Transaction succeeded:", reply);
+  } catch (error) {
+    console.error("Transaction failed:", error);
+    throw error;
+  }
+}
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+const getTxsReports = async (feedback, message, server) => {
+  try {
+    const result = await yeetTx(feedback, server);
+    console.log(`${message}: Transaction succeeded`, result);
+    return result;
+  } catch (error) {
+    console.error(`${message}: Transaction failed`, error);
+  }
```

Comment:
I don't think this function really adds much for us. I do like the idea of logging this status `message` along the way, but I think you're probably better off modifying the `yeetTx()` function itself.